app/code/community/ShipStream/Sync/Model/Order/Shipment/Api.php

```diff
@@ -124,4 +131,104 @@ public function createWithTracking($orderIncrementId, $data)
+            foreach ($package['items'] as $item){
+                $key = $orderItems[$item['sku']];
+                if(isset($itemShippedQty[$key])) {
+                    $itemShippedQty[$key] = bcadd($itemShippedQty[$key], $item['order_item_qty'],4);
+                }
+                else {
+                    $itemShippedQty[$key] = floatval($item['order_item_qty']);
+                }
+            }
+        }
+
+        //discard items that are partially shipped
+        foreach ($itemShippedQty as $item_id => $ordered_qty) {
+            $itemShippedQty[$item_id] = floor($itemShippedQty[$item_id]);
+            if($itemShippedQty[$item_id] == 0){
+                unset($itemShippedQty[$item_id]);
+            }
+
+        }
+
+        return $itemShippedQty;
+    }
+
+    /**
+     * Prepare shipment comment data from Shipstream shipment packages
+     * @param $order
+     * @param $data
+     * @return string
+     */
+    protected function _getCommentsData($order, $data): string
+    {
+        $orderComments = [];
+
+        //get Item name & SKU from magento order items
+        $orderItemsData = $order->getAllItems();
+        foreach ($orderItemsData as $orderItem){
+            $orderComments[$orderItem->getSku()]['sku'] =$orderItem->getSku();
+            $orderComments[$orderItem->getSku()]['name'] = $orderItem->getName();
+        }
+
+        //get lot data of order items
+        foreach ($data['items'] as $item){
+            if(isset($orderComments[$item['sku']])){
+                foreach ($item['lot_data'] as $lot_data)
+                    $orderComments[$item['sku']]['lotdata'][] = $lot_data;
+            }
+        }
+
+        //get collected data of packages from shipment packages
+        foreach ($data['packages'] as $package) {
+            //Shipstream internal order_item_id & SKU to map that with Magento SKU for collected data
+            $orderItems = [];
+            foreach ($package['items'] as $item) {
+                $orderItems[$item['order_item_id']] = $item['sku'];
+            }
+
+            //adding package data value under relevant Order Item
+            foreach ($package['package_data'] as $packageData) {
+                if (isset($orderItems[$packageData['order_item_id']])) {
+                    $sku = $orderItems[$packageData['order_item_id']];
+                    $orderComments[$sku]['collected_data']['label'] = $packageData['label'];
+                    $orderComments[$sku]['collected_data']['value'] = $packageData['value'];
+                }
+            }
+        }
+
+        //format  array to discard indexes
+        $comments = [];
+        foreach ($orderComments as $orderComment){
+            $comments[] = $orderComment;
+        }
+        //format comments data into yaml format if yaml plugin is configured
+        if(function_exists('yaml_emit')) {
+            return yaml_emit($comments);
+        } else {
+            return json_encode($comments);
```

Comment:
Make it more human-readable:

```suggestion
            return json_encode($comments, JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES | JSON_UNESCAPED_UNICODE);
```